tests/topotato/topotato/exabgp/_exabgp.py

```diff
@@ -0,0 +1,220 @@
+
+                    with open(env_dir, "w", encoding="UTF-8") as env_file:
+                        env_file.write(self._cmdobj._env)
+
+                    return
+
+                script_path = Path(__file__).resolve().parent
+
+                template_dir = os.path.join(
+                    script_path, "template", "exabgp.env.jinja2"
+                )
+
+                with open(template_dir, "r", encoding="UTF-8") as template_env, open(
+                    env_dir, "w", encoding="UTF-8"
+                ) as env_file:
+                    template = jenv.from_string(template_env.read())
+                    env = template.render(
+                        {"log_destination": os.path.join(tempdir, "exabgp.log")}
+                    )
+
+                    env_file.write(env)
+
+            def create_config():
+
+                template = jenv.from_string(self._cmdobj._conf)
+
+                conf = template.render({"routers": self._cmdobj._all_routers})
+
+                conf_dir = os.path.join(tempdir, "conf.ini")
+
+                with open(conf_dir, "w", encoding="UTF-8") as env_file:
+                    env_file.writelines(conf)
+
+            create_exabgp_folders()
+            create_cli_files()
+            create_env()
+            create_config()
+
+            return tempdir
+
+        def validate(self):
+            router = self.instance.routers[self._rtr.name]
+            path = self._cmdobj.path
+            self._cmdobj.proc_cli = router.popen(
+                [
+                    "exabgp",
+                    "--debug",
+                    "--validate",
+                    os.path.join(path, "conf.ini"),
+                    "--root",
+                    path,
+                    "--env",
+                    os.path.join(path, "exabgp.env"),
+                ],
+                cwd=os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+            )
+
+            self.is_cli_ok()
+            
+        def run(self):
+            router = self.instance.routers[self._rtr.name]
+            path = self._cmdobj.path
+            self._cmdobj.proc = router.popen(
+                [
+                    "exabgp",
+                    "--debug",
+                    os.path.join(path, "conf.ini"),
+                    "--root",
+                    path,
+                    "--env",
+                    os.path.join(path, "exabgp.env"),
+                ],
+                cwd=os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+            )
+
+            self.is_bgp_daemon_running()
+
+    class Execute(Action):
+        # pylint: disable=consider-using-with
+        def __call__(self):
+
+            self.is_bgp_daemon_running()
+
+            router = self.instance.routers[self._rtr.name]
+            path = self._cmdobj.path
+
+            self._cmdobj.proc_cli = router.popen(
+                ["exabgpcli", "--root", path, self._cmdobj._cmd],
```

Comment:
found the problem on my laptop - `exabgpcli` needs the `--env` parameter too, with the same env file as the exabgp daemon. Then it (correctly) doesn't read `/etc/exabgp/exabgp.env`

Reply:
Thanks, will fix that